packages/reducers/src/combineReducers.test.js

```diff
@@ -0,0 +1,99 @@
+import { createStore } from 'redux';
+import combineReducers from './combineReducers';
+
+describe('combineReducers', () => {
+	const foo = (state = {}) => state;
+	const bar = (state = {}) => state;
+	const ACTION = { type: 'ACTION' };
+
+	it('returns a composite reducer that maps the state keys to given reducers', () => {
+		const reducer = combineReducers({
+			counter: (state = 0, action) => (action.type === 'increment' ? state + 1 : state),
+			stack: (state = [], action) => (action.type === 'push' ? [...state, action.value] : state),
+		});
+		const s1 = reducer(undefined, { type: 'increment' });
+		expect(s1).toEqual({ counter: 1, stack: [] });
+		const s2 = reducer(s1, { type: 'push', value: 'a' });
+		expect(s2).toEqual({ counter: 1, stack: ['a'] });
+	});
+
+	it('ignores all props which are not a function', () => {
+		// we double-cast because these conditions can only happen in a javascript setting
+		const reducer = combineReducers({
+			fake: true,
+			broken: 'string',
+			another: { nested: 'object' },
+			stack: (state = []) => state,
+		});
+		expect(Object.keys(reducer(undefined, { type: 'push' }))).toEqual(['stack']);
+	});
+
+	it('does not have referential equality if one of the reducers changes something', () => {
```

Comment:
did we get rid of the test where we check that we have maintained referential equality?

pls return it, that's important.


```
	it('maintains referential equality if the reducers it is combining do', () => {
			const reducer = combineReducers({
				child1(state = {}) {
					return state;
				},
				child2(state = {}) {
					return state;
				},
				child3(state = {}) {
					return state;
				},
			});
			const initialState = reducer(undefined, { type: '@@INIT' });
			expect(reducer(initialState, { type: 'FOO' })).toBe(initialState);
```

found it